operators/src/pro/cache/cache_chunks.rs

```diff
@@ -0,0 +1,398 @@
+        impl CacheElementSubType for $g {
+            type CacheElementType = FeatureCollection<$g>;
+
+            fn insert_element_into_landing_zone(
+                landing_zone: &mut LandingZoneQueryFeatures,
+                element: Self::CacheElementType,
+            ) -> Result<(), super::error::CacheError> {
+                match landing_zone {
+                    LandingZoneQueryFeatures::$variant(v) => {
+                        v.push(element);
+                        Ok(())
+                    }
+                    _ => Err(super::error::CacheError::InvalidTypeForInsertion),
+                }
+            }
+
+            fn create_empty_landing_zone() -> LandingZoneQueryFeatures {
+                LandingZoneQueryFeatures::$variant(Vec::new())
+            }
+
+            fn result_stream(
+                cache_elements_container: &CachedFeatures,
+                query: &VectorQueryRectangle,
+            ) -> Option<CacheChunkStream<$g>> {
+                if let TypedCacheChunkStream::$variant(v) =
+                    cache_elements_container.chunk_stream(query)
+                {
+                    Some(v)
+                } else {
+                    None
+                }
+            }
+        }
+    };
+}
+impl_cache_element_subtype_magic!(NoGeometry, NoGeometry);
+impl_cache_element_subtype_magic!(MultiPoint, MultiPoint);
+impl_cache_element_subtype_magic!(MultiLineString, MultiLineString);
+impl_cache_element_subtype_magic!(MultiPolygon, MultiPolygon);
+
+/// Our own tile stream that "owns" the data (more precisely a reference to the data)
+#[pin_project(project = CacheChunkStreamProjection)]
+pub struct CacheChunkStream<G> {
+    data: Arc<Vec<FeatureCollection<G>>>,
+    query: VectorQueryRectangle,
+    idx: usize,
+}
+
+impl<G> CacheChunkStream<G>
+where
+    G: Geometry,
+    FeatureCollection<G>: FeatureCollectionInfos,
+{
+    pub fn new(data: Arc<Vec<FeatureCollection<G>>>, query: VectorQueryRectangle) -> Self {
+        Self {
+            data,
+            query,
+            idx: 0,
+        }
+    }
+
+    pub fn element_count(&self) -> usize {
+        self.data.len()
+    }
+}
+
+impl<G: Geometry> Stream for CacheChunkStream<G>
+where
+    FeatureCollection<G>: CacheElementHitCheck,
+    G: ArrowTyped,
+{
+    type Item = Result<FeatureCollection<G>>;
+
+    fn poll_next(
+        mut self: Pin<&mut Self>,
+        _cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        let CacheChunkStreamProjection { data, query, idx } = self.as_mut().project();
+
+        // return the next tile that is contained in the query, skip all tiles that are not contained
+        for i in *idx..data.len() {
+            let chunk = &data[i];
+            if chunk.cache_element_hit(query) {
+                // TODO: we really should cache the elements bbox somewhere
+                let Ok(chunk) = chunk.filter_cache_element_entries(query) else {
```

Comment:
where do you merge the chunks back to the proper chunk size?

Reply:
a0c13c0ab0e1ffac8f290d04616944dbb0e36cc0 
it is not really easy to do this... Adding it to the cache would poisonthe cache with the chunk merger stuff and context information. Adding a new "Wrapper" FeatureChunkMergerQueryProcessor does not work because it requires all streams to be FusedSteam and somehow that does not work. So i just added a trait that has a wrap method....